src/util/coordinate_calculation.cpp

```diff
@@ -275,6 +275,142 @@ double circleRadius(const Coordinate C1, const Coordinate C2, const Coordinate C
         return std::numeric_limits<double>::infinity();
 }
 
+/// Newton-based Taubin circle algebraic fit in a plain.
+/// Chernov, N. Circular and Linear Regression: Fitting Circles and Lines by Least Squares, 2010
+/// Chapter 5.10 Implementation of the Taubin fit, p. 126
+/// Original code http://people.cas.uab.edu/~mosya/cl/CircleFitByTaubin.cpp
+bool circleCenterTaubin(const std::vector<Coordinate> &coords,
+                        std::pair<double, double> &center,
+                        double &estimated_radius)
+{
+    // Compute mean and covariances with the two-pass algorithm
+    double meanX = 0., meanY = 0.;
+    for (const auto &c : coords)
+    {
+        // Use the equirectangular projection
+        const auto x = static_cast<double>(toFloating(c.lon));
+        const auto y = static_cast<double>(toFloating(c.lat));
+        meanX += x;
+        meanY += y;
+    }
+    meanX /= coords.size();
+    meanY /= coords.size();
```

Comment:
This is UB when `coords` is empty, we should assert that precondition or add a guard at the top.